src/segtree.rs

```diff
@@ -1 +1,161 @@
+use crate::internal_bit::ceil_pow2;
 
+pub trait Monoid {
+    type S: Copy;
```

Comment:
We will possibly want to use non `Copy` types such as matrix for `S`.

Reply:
Then should we use `Clone` instead?

Reply:
Yes, that is what I meant. `identit`ies and `binary_operation`s will be inlined.